pkg/s3wrapper/client.go

```diff
@@ -382,3 +390,28 @@ func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]st
 	}
 	return objects, nil
 }
+
+func (c *S3Client) ExtractFilesFromISOAndUpload(ctx context.Context, isoFilePath, isoObjectName string) error {
+	log := logutil.FromContext(ctx, c.log)
+	err := ExtractFilesFromISOAndUploadStream(ctx, log, isoFilePath, isoObjectName, c)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (c *S3Client) DownloadPXEArtifact(ctx context.Context, fileType string) (io.ReadCloser, string, int64, error) {
+	objectName := strings.TrimSuffix(BaseObjectName, ".iso") + "." + fileType
+	reader, contentLength, err := c.Download(ctx, objectName)
+	return reader, objectName, contentLength, err
+}
+
+func (c *S3Client) GetS3PXEArtifactURL(fileType string) string {
+	objectName := strings.TrimSuffix(BaseObjectName, ".iso") + "." + fileType
+	if c.cfg.S3EndpointURL == "" {
```

Comment:
When is this true?

Reply:
If using AWS S3, there is no need to provide an endpoint, just region.

Reply:
Not need, maybe, but it is provided to us in the cloud environments so maybe better to just rely on those?